Type checker errors should be reported in source/compilation order

## Summary
Errors from imported modules should be reported before downstream errors in the importing file. Currently, errors can appear in a confusing order where a symptom (e.g., "undefined function") is shown instead of or before the root cause (e.g., type error in the imported module).

## Problem
When an imported module has type errors, the user may see misleading downstream errors first:

1. `deserializer.ez` has type error on line 29: `invalid cast target type: float64`
2. `main.ez` imports and uses a function from that module
3. User sees: `undefined function 'codec.deserialize_to_u64'`
4. User is confused - the function DOES exist

The root cause error (type mismatch in deserializer.ez) should be shown first, not the symptom.

## Expected Behavior
Errors should be reported in compilation order:
1. Errors in imported modules first (in the order modules are processed)
2. Within each file, errors in source line order
3. Downstream/dependent errors last

Example of correct ordering:
```
error[E3001]: invalid cast target type: float64
  --> src/codec/deserializer.ez:29:12

error[E4002]: undefined function 'codec.deserialize_to_u64'  
  --> main.ez:28:25
  = note: this may be caused by the error above
```

## Current Behavior
Errors appear in arbitrary order, or root cause errors are swallowed entirely (see #726), leaving users with confusing symptom-only errors.

## Related
- #726 - Parser errors in imported modules are silently ignored

--- a/pkg/errors/format.go
+++ b/pkg/errors/format.go
@@ -5,6 +5,7 @@ package errors
 
 import (
 	"fmt"
+	"sort"
 	"strings"
 )
 
@@ -89,18 +90,41 @@ func FormatError(err *EZError) string {
 	return sb.String()
 }
 
+// sortErrors sorts errors by file path, then line number, then column
+func sortErrors(errors []*EZError) []*EZError {
+	sorted := make([]*EZError, len(errors))
+	copy(sorted, errors)
+	sort.Slice(sorted, func(i, j int) bool {
+		// Sort by file path first
+		if sorted[i].File != sorted[j].File {
+			return sorted[i].File < sorted[j].File
+		}
+		// Then by line number
+		if sorted[i].Line != sorted[j].Line {
+			return sorted[i].Line < sorted[j].Line
+		}
+		// Then by column
+		return sorted[i].Column < sorted[j].Column
+	})
+	return sorted
+}
+
 // FormatErrorList formats multiple errors and warnings
 func FormatErrorList(el *EZErrorList) string {
 	var sb strings.Builder
 
+	// Sort errors by file and line number before printing
+	sortedErrors := sortErrors(el.Errors)
+	sortedWarnings := sortErrors(el.Warnings)
+
 	// Print all errors first
-	for _, err := range el.Errors {
+	for _, err := range sortedErrors {
 		sb.WriteString(FormatError(err))
 		sb.WriteString("\n")
 	}
 
 	// Then print warnings
-	for _, warn := range el.Warnings {
+	for _, warn := range sortedWarnings {
 		sb.WriteString(FormatError(warn))
 		sb.WriteString("\n")
 	}

